graphs/minimum_effort_path.py

```diff
@@ -15,4 +16,16 @@ def min_effort_path(heights):
         int
             minimum effort required to navigate the path from (0, 0) to heights[rows - 1][columns - 1]
     """
-    pass
+    if not heights or not heights[0]:
+        return 0
+    rows = len(heights)
+    columns = len(heights[0])
+    dp = [[float("inf") for _ in range(columns)] for _ in range(rows)]
+    dp[0][0] = 0
+    for i in range(rows):
+        for j in range(columns):
+            if i - 1 >= 0:
+                dp[i][j] = min(dp[i][j], max(dp[i-1][j], abs(heights[i][j]-heights[i-1][j])))
```

Comment:
I see what you're doing here, and it's quite clever. However, I would recommend using Dijkstra's instead and using a priority queue/heap to choose which cell you visit next and drive your calculations.

The issue you're running into here is that for each cell, you calculate whether the height differential between the current cell in your height matrix and the `[i-1][j]`th cell (neighbor to the left) is smaller than any height differential you've calculated thus far. You try and bring forward whether the max height differential in a neighboring cell is larger than the max height differential calculated for the current cell so you can get the max elevation change along the entire path instead of just for the current cell. 

However, you're iterating through your cells in 'switchbacks', following row 0 from column 0 to its end, then the next row and the next. This isn't necessarily the minimum effort path you want to follow, so sometimes you're bringing forward height differentials/elevation changes that aren't actually relevant to the 'minimum effort path'. 

You've found a couple gaps in our test coverage as well, but you'll want to check all neighboring cells (up, left, right, down) not just the upper and left neighbors. It is possible that a path can require you to go to the `i+1`th or `j+1`th cell.